File: Array/11_20/18Solution.py
def maxScore(cardPoints,k):
        sub_sum = answer = sum(cardPoints[:k])
        left = k-1
        right = len(cardPoints) - 1
        for i in range(k):
            sub_sum += cardPoints[right] - cardPoints[left] 
            left -= 1
            right -= 1
            answer = max(answer, sub_sum)
        return answer
cardPoints=[int(x.strip()) for x in input().split(",")]
k=int(input())

File: Array/11_20/18Solution.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int subsum=0;
        for(int i=0;i<k;i++){
            subsum+=cardPoints[i];
        }
        int left=k-1,right=cardPoints.length-1;
        int max=subsum;
        for(int i=0;i<k;i++){
            subsum+=(cardPoints[right]-cardPoints[left]);
            right--;
            left--;
            if(max<subsum){
                max=subsum;
            }
        }
        return max;
    }
}
